<launch>
    <arg name="camera_2d_wide_enabled"/>
    <arg name="output_directory" default="$(env HOME)/.ros/t-top/recordings"/>

    <node pkg="recorders" exec="video_recorder_node.py" name="video_recorder_camera_3d_node">
        <param name="output_directory" value="$(var output_directory)"/>
        <param name="filename_prefix" value="camera_3d_"/>

        <param name="video_stream_name" value="camera_3d"/>
        <param name="video_stream_format" value="rgb8"/>
        <param name="video_stream_width" value="1280"/>
        <param name="video_stream_height" value="720"/>
        <param name="video_stream_framerate" value="15"/>
        <param name="video_stream_codec" value="h264"/>
        <param name="video_stream_bitrate" value="2000000"/>
        <param name="video_stream_delay_s" value="0.35"/>

        <param name="audio_stream_name" value="odas"/>
        <param name="audio_stream_format" value="signed_16"/>
        <param name="audio_stream_channel_count" value="1"/>
        <param name="audio_stream_sampling_frequency" value="16000"/>
        <param name="audio_stream_codec" value="aac"/>

        <remap from="video_camera_3d" to="camera_3d/color/image_raw"/>
        <remap from="audio_odas" to="audio_input_signed_16_16000"/>
        <remap from="video_recorder/filter_state" to="video_recorder_camera_3d/filter_state"/>
    </node>

    <node if="$(var camera_2d_wide_enabled)" pkg="recorders" exec="video_recorder_node.py" name="video_recorder_camera_2d_wide_node">
        <param name="output_directory" value="$(var output_directory)"/>
        <param name="filename_prefix" value="camera_2d_wide_"/>

        <param name="video_stream_name" value="camera_2d_wide"/>
        <param name="video_stream_format" value="bgr8"/>
        <param name="video_stream_width" value="1280"/>
        <param name="video_stream_height" value="720"/>
        <param name="video_stream_framerate" value="5"/>
        <param name="video_stream_codec" value="h264"/>
        <param name="video_stream_bitrate" value="2000000"/>
        <param name="video_stream_delay_s" value="0.0"/>

        <param name="audio_stream_name" value="raw"/>
        <param name="audio_stream_format" value="signed_32"/>
        <param name="audio_stream_channel_count" value="16"/>
        <param name="audio_stream_sampling_frequency" value="16000"/>
        <param name="audio_stream_codec" value="aac"/>

        <remap from="video_camera_2d_wide" to="camera_2d_wide/image_rect"/>
        <remap from="audio_raw" to="audio_input_before_ego_noise_signed_32_16000"/>
        <remap from="video_recorder/filter_state" to="video_recorder_camera_2d_wide/filter_state"/>
    </node>
</launch>
